File: 82-remove-duplicates-from-sorted-list-ii/remove-duplicates-from-sorted-list-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        Set<Integer> set1=new LinkedHashSet<>();
        Set<Integer> set2=new LinkedHashSet<>();
        ListNode temp=head;
        while (temp!=null){
            if(!set1.add(temp.val)){
                set2.add(temp.val);
            }
            temp=temp.next;
        }
        set1.removeAll(set2);
        ListNode dummy=new ListNode(0);
        ListNode t=dummy;
        for(int s:set1){
            t.next=new ListNode(s);
            t=t.next;
        }
        return dummy.next;

    }
}

File: 82-remove-duplicates-from-sorted-list-ii/remove-duplicates-from-sorted-list-ii.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        set1=set()
        set2=set()
        temp=head
        while temp:
            if temp.val in set1:
                set2.add(temp.val)
            else:
                set1.add(temp.val)
            temp=temp.next
        
        temp=head

        dummy=ListNode(0)
        t=dummy
        while temp:
            if temp.val not in set2:
                t.next=ListNode(temp.val)
                t=t.next
            temp=temp.next
        
        return dummy.next